showcase/app/components/page-components/modal/code-fragments/with-trigger.gts

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { on } from '@ember/modifier';
+import { hash } from '@ember/helper';
+
+import { HdsModal } from '@hashicorp/design-system-components/components';
+
+// types
+import type { HdsModalSignature } from '@hashicorp/design-system-components/components/hds/modal/index';
+
+export interface CodeFragmentWithTriggerSignature {
+  Args: {
+    triggerText?: string;
+    modalId: string;
+    size?: HdsModalSignature['Args']['size'];
+    color?: HdsModalSignature['Args']['color'];
+    isDismissDisabled?: HdsModalSignature['Args']['isDismissDisabled'];
+    returnFocusTo?: HdsModalSignature['Args']['returnFocusTo'];
+    class?: string;
+    onOpen?: HdsModalSignature['Args']['onOpen'];
+    onClose?: HdsModalSignature['Args']['onClose'];
+  };
+  Blocks: {
+    trigger: [{ openModal: () => void }];
+    modal: [
+      HdsModalSignature['Blocks']['default'][0] & {
+        close: () => void;
+        isOpen: boolean;
+      },
+    ];
+  };
+}
+
+export default class CodeFragmentWithTrigger extends Component<CodeFragmentWithTriggerSignature> {
+  @tracked isModalOpen = false;
+
+  openModal = () => {
+    this.isModalOpen = true;
+    this.args.onOpen?.();
+  };
+
+  closeModal = () => {
+    this.isModalOpen = false;
+    // Create a mock event for the onClose callback
+    const mockEvent = new Event('close');
+    this.args.onClose?.(mockEvent);
+  };
+
+  <template>
+    {{#if (has-block "trigger")}}
+      {{yield (hash openModal=this.openModal) to="trigger"}}
+    {{else}}
+      <button type="button" {{on "click" this.openModal}}>
+        {{@triggerText}}
+      </button>
+    {{/if}}
+
+    {{#if this.isModalOpen}}
+      <HdsModal
+        id={{@modalId}}
+        class={{@class}}
```

Comment:
[thought] For the HTML `id` and `class` maybe you could rely on the `...attributes` operator (the downside is that you can't define the `id` as required)

Reply:
Good call. I replaced them in 0141a77